distributed/protocol/utils.py

```diff
@@ -67,6 +67,9 @@ def merge_frames(header, frames):
     if all(len(f) == l for f, l in zip(frames, lengths)):
         return frames
 
+    if len(lengths) == 1:
+        return [b"".join(map(ensure_bytes, frames))]
```

Comment:
Or do you mean you would like to see something like this?

```suggestion
    if len(lengths) == 1 and len(frames) == 1:
        return frames
```

Reply:
Yes, `return ensure_bytes(frames[0])` (although I don't see how it could not be bytes).

Reply:
I think we have `bytearray` in the TCP code path and NumPy arrays in the UCX code path. So coercing to `bytes` can make sense. Use a `list` to wrap it as well since that seems to be expected (based on related testing before).

```suggestion
    if len(lengths) == 1 and len(frames) == 1:
        return [ensure_bytes(frames[0])]
```

Reply:
yep, agree

Reply:
That said, I guess `merge_frames` is collecting some frames unaltered, some as `memoryview`s, and some as `bytes`. So maybe we can just pass `frames` through as-is? Would potentially save us a copy.

Reply:
I don't know about that, depends if all callers explicitly expect bytes or not.

Reply:
Just for context, here are the lines I'm looking at:

1. [Preserve frame]( https://github.com/dask/distributed/blob/85620bf43d67807e60cdcb45d18270f131a1cb14/distributed/protocol/utils.py#L80 )
2. [Convert to `memoryview`]( https://github.com/dask/distributed/blob/85620bf43d67807e60cdcb45d18270f131a1cb14/distributed/protocol/utils.py#L85 )
3. [Convert to `bytes` (if there are multiple frames)]( https://github.com/dask/distributed/blob/85620bf43d67807e60cdcb45d18270f131a1cb14/distributed/protocol/utils.py#L90 )

My guess is the only expectation is that frames must be `bytes`-like, which they are when inputted or when coerced to `bytes` or `memoryviews`. So I think we are safe to pass the frames as-is.

Reply:
Maybe a good use case for typing - even though I'm not sure how you say "bytes-like".

Reply:
(that was just musing, not suggesting a change)

Reply:
Good question. I guess it's an open issue ( https://github.com/python/typing/issues/593 ).